File: banda/IvanRC-banda-Java/src/main/java/modelsBandaMusicos/Teclista.java
package modelsBandaMusicos;

public class Teclista extends Musico implements ITeclista {

    String nombre;
    Double salario = 1500.00;
    int añosExperiencia;
    int cantidadTeclas;

    public Teclista(String nombre, int añosExperiencia, int cantidadTeclas) {
        super(nombre, añosExperiencia);
        this.cantidadTeclas = cantidadTeclas;
    }

    public int getCantidadTeclas() {
        return cantidadTeclas;
    }

    public double recalcularSalario(double salario) {
        return ITeclista.super.recalcularSalario(salario);
    }
}

File: banda/SergioSF-Conservatorio-Kotlin/src/main/kotlin/models/Teclista.kt
package models

class Teclista(nombre: String, apellidos: String, edad: Int, anosExperiencia: Int) : Musico(
    nombre, apellidos, edad,
    anosExperiencia
) {
    override var instrumento: Instrumento = Instrumento.TECLADO
    override var especialidad: Especialidad = Especialidad.TECLISTA

    override var salario: Int = (super.salario * 1.30).toInt()
    override var representacion: Int = 10

    override fun interpretar() {
        tocarTeclado()
    }

    override fun presentarse() {
        println("Hola soy $nombre, $apellidos, tengo $edad años y llevo $anosExperiencia años siendo músico. Mi instrumento ${Instrumento.TROMPETA}")
    }

    fun tocarTeclado() {
        println("")
    }

    override fun toString(): String {
        return println("$salario").toString()
    }
}
